src/main/java/com/ibanity/apis/client/http/impl/IbanityHttpClientImpl.java

```diff
@@ -0,0 +1,113 @@
+        ibanityResponseHandler = new IbanityResponseHandler();
+    }
+
+    @Override
+    public String get(URI path) {
+        return get(path, null, newHashMap());
+    }
+
+    @Override
+    public String get(URI path, String customerAccessToken) {
+        return get(path, customerAccessToken, newHashMap());
+    }
+
+    @Override
+    public String get(URI path, String customerAccessToken, Map<String, String> additionalHeaders) {
+        try {
+            HttpGet httpGet = new HttpGet(path);
+            addHeaders(customerAccessToken, additionalHeaders, httpGet);
+            return httpClient.execute(httpGet, ibanityResponseHandler);
+        } catch (IOException exception) {
+            throw new RuntimeException("An error occurred while connecting to Ibanity", exception);
+        }
+    }
+
+    @Override
+    public String post(URI path, Object requestApiModel) {
+        return post(path, requestApiModel, null, newHashMap());
+    }
+
+    @Override
+    public String post(URI path, Object requestApiModel, String customerAccessToken) {
+        return post(path, requestApiModel, customerAccessToken, newHashMap());
+    }
+
+    @Override
+    public String post(URI path, Object requestApiModel, String customerAccessToken, Map<String, String> additionalHeaders) {
+        try {
+            HttpPost httpPost = new HttpPost(path);
+            addHeaders(customerAccessToken, additionalHeaders, httpPost);
+            httpPost.setEntity(createEntityRequest(objectMapper().writeValueAsString(requestApiModel)));
+            return httpClient.execute(httpPost, ibanityResponseHandler);
+        } catch (IOException exception) {
+            throw new RuntimeException("An error occurred while connecting to Ibanity", exception);
+        }
+    }
+
+    @Override
+    public String delete(URI path) {
+        return delete(path, null, Collections.emptyMap());
+    }
+
+    @Override
+    public String delete(URI path, String customerAccessToken) {
+        return delete(path, customerAccessToken, Collections.emptyMap());
+    }
+
+    @Override
+    public String delete(URI path, String customerAccessToken, Map<String, String> additionalHeaders) {
+        try {
+            HttpDelete httpDelete = new HttpDelete(path);
+            addHeaders(customerAccessToken, additionalHeaders, httpDelete);
+            return httpClient.execute(httpDelete, ibanityResponseHandler);
+        } catch (IOException exception) {
+            throw new RuntimeException("An error occurred while connecting to Ibanity", exception);
+        }
+    }
+
+    private HttpEntity createEntityRequest(String baseRequest) {
+        return new StringEntity(baseRequest, APPLICATION_JSON);
+    }
+
+    private void addHeaders(String customerAccessToken, Map<String, String> additionalHeaders, HttpRequestBase httpRequestBase) {
+        addAuthorizationHeader(customerAccessToken, httpRequestBase);
+        additionalHeaders.forEach(httpRequestBase::addHeader);
```

Comment:
The goal is to avoid handling null everywhere but instead play with empty collections.
It reduces the risk of NPE and improve code readability by removing the need for null checks and allowing fluent method calls.
I will update the interface with the nonNull annotation.